zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/UserCreatedApplier.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.engine.state.mutable.MutableUserState;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import io.camunda.zeebe.protocol.record.intent.UserIntent;
+
+public class UserCreatedApplier implements TypedEventApplier<UserIntent, UserRecord> {
+
+  private final MutableUserState userState;
+
+  public UserCreatedApplier(final MutableProcessingState processingState) {
+    userState = processingState.getUserState();
+  }
+
+  @Override
+  public void applyState(final long key, final UserRecord value) {
+    userState.create(value);
```

Comment:
❓ We don't use the key here and we don't store it in the state. The key is fully unique to a user. Is the name unique enough so that we don't need to store the key in the state?

Reply:
The `username` prop is intended to be unique so really the key shouldn't need to be stored in the state